src/Tools/dotnet-trace/DiagnosticProfileHandlers/GcPauseHandler.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Tools.Trace.DiagnosticProfileHandlers
+{
+    /// <summary>
+    /// This class handles parsing for GC-triggered Pauses - specifically by parsing the GC/SuspendEEStart, GC/SuspendEEStop, GC/GCStart, GC/GCEnd events.
+    /// </summary>
+    internal class GcPauseHandler : IDiagnosticProfileHandler
+    {
+        private DateTime lastKnownEESuspensionStartTimeStamp;
+        private Dictionary<int, DateTime> GCStartTimeCache; // Cache of start-times for each GC to compute GC pause times.
+
+        public GcPauseHandler()
+        {
+            GCStartTimeCache = new Dictionary<int, DateTime>();
+        }
+
+        /// <summary>
+        /// Adds the GC Handler
+        /// NOTE: This handler is not thread-safe. If we ever modify it to be used from multiple threads, we need to modify the read/writes to 
+        /// the GCStartTimeCache dictionary to be thread-safe.
+        /// </summary>
+        /// <param name="source"></param>
+        public void RegisterHandler(EventPipeEventSource source)
+        {
+            source.Clr.GCStart += (GCStartTraceData data) =>
+            {
+                GCStartTimeCache[data.Count] = data.TimeStamp;
+                Console.WriteLine($"[CLR-GC|GCStart|{data.TimeStamp.ToString("s", CultureInfo.InvariantCulture)}] Count={data.Count};Reason={GetGCReason(data.Reason)};Type={GetGCType(data.Type)}");
```

Comment:
Would it be possible to use an abstraction for the logging (such as ILogger)? This will be especially useful in cases where we want to unify code with dotnet-monitor.